removed reset pin dependency in case of BG95

n3rd4i opened the pull request:
As described in the bug ticket, the BG95 modem does not proper initialize. There are 3 issues with the current implementation (for which I will try to address in 3 separate commits in this PR):
* [1/3] The reset pin is considered mandatory, but internally BG95 does not make use of such pin (it is mapped to power pin)
* [2/3] Using bg9x driver with bg95 specifically has the issue that modem replies with RDY when it's not fully initialized and AT commands sent to the modem after this point would result in modem reply of "ERROR". In documentation it is mentioned that "APP RDY" has to be considered instead.
* [3/3] because reset pin is mapped to power pin, a reset on the device (e.g. external reset of the board using the bg95 modem) would not necessarily result in a "cold-start" initialization but could result in a power-off (depending on the previous state)

Commit:
drivers: modem: quectel-bg9x: fix for bg95 pinout

The BG95 pin configuration does not internally ever use the reset pin.
Because of this, there is no need to make reset pin mandatory.
Commit removes reset pin dependency [e.g. in case of BG95].

Signed-off-by: Andrei Hutanu <andrei.hutanu.i@gmail.com>

rerickson1 commented:
Please fix your commit message to contain a sign-off field and a description of why you are making the change.

rerickson1 commented on drivers/modem/Kconfig.quectel-bg9x:
```diff
@@ -54,6 +54,14 @@ config MODEM_QUECTEL_BG9X_PASSWORD
 	  context.  This value is specific to the network provider and may
 	  need to be changed.
 
+config MODEM_QUECTEL_BG9X_UNSOL_RDY
+	string "Unsolicited response string expected from the modem"
+	default "RDY"
+	help
+	  This string is used to determine when modem powered and ready
+	  to accept AT commands. If this value is not matching the modem
+	  response, the init will fail with timeout.
+
```
Why does this need to be a kconfig value? This should never be changing. The AT command responses are defined and do not change.

n3rd4i replied:
it does not have to be, but otherwise there has to be a differentiation between MODEM_QUECTEL_BG9X and presumably MODEM_QUECTEL_BG9x_BG95. I don't know how else this could be accomplished?

n3rd4i replied:
![image](https://user-images.githubusercontent.com/7346660/233668578-8d198298-b3a4-47f2-8080-38f508facfaf.png)
Like I've mentioned in the original ticket, as "the issue". (screenshot from `Quectel_BG95BG77BG600L_Series_AT_Commands_Manual_V2.0-2.pdf`)

Commit:
drivers: modem: quectel-bg9x: mdm unsol ready

Using bg9x driver with bg95 specifically has the issue that BG95
modem replies with "RDY" when it's not fully initialized and AT
commands sent to the modem after this point would result in modem
reply of "ERROR". This commit adds a new prj.conf field in the build
configuration for custom unsolicited app ready string, with
the default being BG96 one ("RDY").

Signed-off-by: Andrei Hutanu <andrei.hutanu.i@gmail.com>

Commit:
drivers: modem: quectel-bg9x: fix reset BG95 mdm

In case of BG95 where there is no reset pin, reseting the board
which uses a BG95 modem could result in powering-off the modem
(as oposed to "reset" the modem). This commit is an attempt
to handle this case and re-power the modem.

Signed-off-by: Andrei Hutanu <andrei.hutanu.i@gmail.com>
